fendermint/vm/interpreter/src/fvm/broadcast.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use anyhow::{anyhow, bail, Context};
+use ethers::types as et;
+use fvm_shared::{address::Address, chainid::ChainID, econ::TokenAmount, BLOCK_GAS_LIMIT};
+use num_traits::Zero;
+use tendermint_rpc::Client;
+
+use fendermint_crypto::SecretKey;
+use fendermint_rpc::message::GasParams;
+use fendermint_rpc::query::QueryClient;
+use fendermint_rpc::tx::{CallClient, TxClient, TxCommit};
+use fendermint_rpc::{client::FendermintClient, message::MessageFactory};
+use fendermint_vm_message::query::FvmQueryHeight;
+
+/// Broadcast transactions to Tendermint.
+///
+/// This is typically something only active validators would want to do
+/// from within Fendermint as part of the block lifecycle, for example
+/// to submit their signatures to the ledger.
+///
+/// The broadcaster encapsulates the tactics for figuring out the nonce,
+/// the gas limit, potential retries, etc.
+#[derive(Clone)]
+pub struct Broadcaster<C> {
+    client: FendermintClient<C>,
+    secret_key: SecretKey,
+    addr: Address,
+    gas_fee_cap: TokenAmount,
+    gas_premium: TokenAmount,
+}
+
+impl<C> Broadcaster<C>
+where
+    C: Client + Clone + Send + Sync,
+{
+    pub fn new(
+        client: C,
+        secret_key: SecretKey,
+        gas_fee_cap: TokenAmount,
+        gas_premium: TokenAmount,
+    ) -> Self {
+        let client = FendermintClient::new(client);
+        // TODO: We could use f410 addresses to send the transaction, but the `MessageFactory` assumes f1.
```

Comment:
Hm, okay, I will think of a way to add support for both. 

I thought that since we discussed that the user who bonds with their Ethereum account to become a validator has to provide a public key they will use to sign blocks, it's not a big ask for them to fund such an f1 account with tokens. But if that's awkward, we can add something to show whether the secret key should be treated as an f410 or f1 address.

Reply:
Created https://github.com/consensus-shipyard/fendermint/issues/283